src/Avalonia.Visuals/Media/TextFormatting/TextFormatterImpl.cs

```diff
@@ -548,5 +602,13 @@ public bool MoveNext()
                 return !(Current is TextEndOfLine);
```

Comment:
`return true`

Reply:
After this change, every `ITextSource` that previously used `EndOfLine` as the end marker should now return null